drivers/npmx/npmx_shell.c

```diff
@@ -2290,690 +2299,481 @@ static int cmd_gpio_status_get(const struct shell *shell, size_t argc, char **ar
 		return 0;
 	}
 
-	int err = 0;
-	uint8_t ldsw_indx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
+	int gpio_index = args_info.arg[1].result.ivalue;
+	npmx_ldsw_gpio_config_t gpio_config = {
+		.gpio = ldsw_gpio_index_convert(gpio_index),
+		.inverted = args_info.arg[2].result.bvalue,
+	};
 
-	if (err != 0) {
-		shell_error(shell, "Error: instance index has to be an integer.");
+	if (gpio_config.gpio == NPMX_LDSW_GPIO_INVALID) {
+		shell_error(shell, "Error: wrong GPIO index.");
 		return 0;
 	}
 
-	if (ldsw_indx >= NPM_LDSW_COUNT) {
-		shell_error(shell, "Error: LDSW instance index is too high: no such instance.");
+	if (!check_pin_configuration_correctness(shell, gpio_index)) {
 		return 0;
 	}
 
-	uint8_t status_mask;
-	npmx_ldsw_t *ldsw_instance = npmx_ldsw_get(npmx_instance, ldsw_indx);
-	npmx_error_t err_code = npmx_ldsw_status_get(ldsw_instance, &status_mask);
-	uint8_t check_mask = ldsw_indx == 0 ? (NPMX_LDSW_STATUS_POWERUP_LDSW_1_MASK |
-					       NPMX_LDSW_STATUS_POWERUP_LDO_1_MASK) :
-					      (NPMX_LDSW_STATUS_POWERUP_LDSW_2_MASK |
-					       NPMX_LDSW_STATUS_POWERUP_LDO_2_MASK);
-
-	if (check_error_code(shell, err_code)) {
-		shell_print(shell, "Value: %d.", !!(status_mask & check_mask));
-	} else {
-		shell_error(shell, "Error: unable to get LDSW status.");
+	npmx_error_t err_code = npmx_ldsw_enable_gpio_set(ldsw_instance, &gpio_config);
+	if (!check_error_code(shell, err_code)) {
+		print_set_error(shell, "GPIO config");
+		return 0;
 	}
 
+	shell_print(shell, "Success: %d %d.", gpio_index, gpio_config.inverted);
 	return 0;
 }
 
-static int cmd_ldsw_mode_set(const struct shell *shell, size_t argc, char **argv)
+static int cmd_ldsw_gpio_get(const struct shell *shell, size_t argc, char **argv)
 {
-	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
-
-	if (npmx_instance == NULL) {
-		shell_error(shell, "Error: shell is not initialized.");
+	args_info_t args_info = { .expected_args = 1,
+				  .arg = {
+					  [0] = { SHELL_ARG_TYPE_UINT32_INDEX, "LDSW" },
+				  } };
+	if (!arguments_check(shell, argc, argv, &args_info)) {
 		return 0;
 	}
 
-	if (argc < 2) {
-		shell_error(shell, "Error: missing LDSW instance index and LDSW mode.");
+	npmx_ldsw_t *ldsw_instance = ldsw_instance_get(shell, args_info.arg[0].result.uvalue);
+	if (ldsw_instance == NULL) {
 		return 0;
 	}
 
-	if (argc < 3) {
-		shell_error(shell, "Error: missing LDSW mode (0 -> LOADSW, 1 -> LDO).");
+	npmx_ldsw_gpio_config_t gpio_config;
+	npmx_error_t err_code = npmx_ldsw_enable_gpio_get(ldsw_instance, &gpio_config);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "GPIO config");
 		return 0;
 	}
 
-	int err = 0;
-	uint8_t ldsw_indx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
-	npmx_ldsw_mode_t mode =
-		(npmx_ldsw_mode_t)(CLAMP(shell_strtoul(argv[2], 0, &err), 0, UINT8_MAX));
+	int8_t gpio_index =
+		(gpio_config.gpio == NPMX_LDSW_GPIO_NC) ? -1 : ((int8_t)gpio_config.gpio - 1);
```

Comment:
If it is casted to `int8_t`, then `gpio_config.gpio - 1` should be enough, it should also work for `NPMX_LDSW_GPIO_NC`